### PR TITLE
Add functionality for AWS DNS framework and route53 & dyn dns provider

### PR DESCRIPTION
**Pull request by mazzystr**
https://trello.com/c/R6yaXLpF/666-aws-dns-hook

Scenario:
-----------------------------------------

As an openshift Ops admin
I want AWS provisioning to include a DNS hook and create the CNAME for the ELB so that I can browse to openshift_master_cluster_public_hostname or openshift_master_cluster_hostname when installer completes.

**Acceptance Criteria**
- provision_install play should deploy all AWS infrastructure
- provision_install play should install OpenShift

Notes:
-----------------------------------------
https://bugzilla.redhat.com/show_bug.cgi?id=1569635
https://bugzilla.redhat.com/show_bug.cgi?id=1569631

### CODE REVIEW

**Comment by DanyC97** [comment]
is there a use case for OCP to use AWS with dyn as dns provider?

File: roles/lib_dyn/meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+galaxy_info:
+  author: Russell Harrison
+  description:  A role to provide the dyn_record module
```
Comment (vrutkovs) [inline roles/lib_dyn/meta/main.yml]: `4:16      error    too many spaces after colon  (colons)`

File: roles/lib_dyn/meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  company: Red Hat, Inc.
+  # If the issue tracker for your role is not on github, uncomment the
+  # next line and provide a value
+  # issue_tracker_url: http://example.com/issue/tracker
+  license: Apache
+  min_ansible_version: 1.9
+  platforms:
+    - name: EL
+      versions:
+       - 7
```
Comment (vrutkovs) [inline roles/lib_dyn/meta/main.yml]: `14:8      error    wrong indentation: expected 8 but found 7  (indentation)`

File: roles/lib_dyn/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  #  - 19
+  #  - 20
+  #  - 21
+  #  - 22
+  # Below are all categories currently available. Just as with
+  # the platforms above, uncomment those that apply to your role.
+  categories:
+    - networking
+dependencies: []
+  # List your role dependencies here, one per line.
```
Comment (vrutkovs) [inline roles/lib_dyn/meta/main.yml]: `26:3      warning  comment not indented like content  (comments-indentation)`

File: roles/lib_dyn/meta/main.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  # Be sure to remove the '[]' above if you add dependencies
+  # to this list.
+  #
+  # No role dependencies at this time. The module contained in this role does
+  # require the dyn python module.
+  # https://pypi.python.org/pypi/dyn
+
```
Comment (vrutkovs) [inline roles/lib_dyn/meta/main.yml]: `33:1      error    too many blank lines (1 > 0)  (empty-lines)`

File: roles/lib_dyn/library/dyn_record.py [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+    # pylint: disable=redefined-builtin, unused-wildcard-import, wildcard-import,
+    # locally-disabled
+    from ansible.module_utils.basic import AnsibleModule
+    from dyn.tm.session import DynectSession
+    from dyn.tm.zones import Zone
+    import dyn.tm.errors
+    import os
+
+except ImportError as error:
+    IMPORT_ERROR = str(error)
```
Comment (vrutkovs) [inline roles/lib_dyn/library/dyn_record.py]: Needs `# pylint: disable=redefined-variable-type` comment before that, as pylint doesn't like redefining var type

Comment (mazzystr) [reply roles/lib_dyn/library/dyn_record.py]: Yes.  It's an Ops blocker.  See [bugzilla 1569631](https://bugzilla.redhat.com/show_bug.cgi?id=1569631)

**Comment by mazzystr** [comment]
Yes.  It's an Ops blocker.  See [bugzilla 1569631](https://bugzilla.redhat.com/show_bug.cgi?id=1569631)

**Comment by mazzystr** [comment]
bot, retest this please

**Comment by mazzystr** [comment]
Is there anything I can do to help this along?

**Comment by mazzystr** [comment]
We found a scenario where this pull won't work.  I'm going to close and submit another.

**Pull request closed without merging**